src/visitor.rs

```diff
@@ -161,6 +154,13 @@ impl<'a> FmtVisitor<'a> {
             visit::FnKind::Closure => None,
         };
 
+        if let Some(ref single_line_fn) = self.rewrite_single_line_fn(&rewrite, &b) {
```

Comment:
Since `rewrite_fn` only handles the function declaration -- not the actual body block -- we'd need to return a bool, or check whether the last character is a closing brace, to know whether to skip `visit_block`. The first solution diverges from the return type of every other `rewrite_X` function, but the second would be fine.